ldap/3.5/modules/ROOT/pages/index.adoc

```diff
@@ -0,0 +1,553 @@
+		doc:name="HTTP Listener config">
+		<http:listener-connection host="127.0.0.1"
+			port="8081" />
+	</http:listener-config>
+
+	<ldap:config name="LDAP_Configuration" doc:name="LDAP Configuration">
+		<ldap:basic-connection authDn="${config.principal.dn}"
+			authPassword="${config.password}" url="${config.url}">
+      <ldap:extended-configurations>
+         <ldap:extended-configuration key="key.name" value="key.value" />
+         </ldap:extended-configuration>
+      </ldap:extended-configurations>
+		</ldap:basic-connection>
+	</ldap:config>
+
+	<flow name="ldap-add-entry-flow">
+		<http:listener config-ref="HTTP_Listener_config" path="/"
+			doc:name="HTTP" />
+
+		<ee:transform doc:name="DataWeave to Create DevOps Group Object">
+			<ee:message>
+				<ee:set-payload><![CDATA[%dw 2.0
+output application/java
+---
+{
+	dn : "ou=DevOpsGroup," ++ attributes.queryParams.dn,
+	ou : "DevOpsGroup",
+	objectclass : ["top", "organizationalUnit"]
+}
+]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+
+		<ldap:add config-ref="LDAP_Configuration"
+			doc:name="Add Group Entry to LDAP Directory" />
+
+
+		<ee:transform doc:name="DataWeave to Create User Object">
+			<ee:message>
+				<ee:set-payload><![CDATA[%dw 2.0
+output application/java
+---
+{
+	dn : "cn=Test User,ou=DevOpsGroup," ++ attributes.queryParams.dn,
+	uid : "testUser",
+	cn : "Test User",
+	sn : "User",
+	userPassword : "test1234",
+	objectclass : ["top", "person", "organizationalPerson", "inetOrgPerson"]
+}
+]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+
+
+		<ldap:add config-ref="LDAP_Configuration"
+			doc:name="Add User Entry to LDAP Directory" />
+
+		<ldap:delete config-ref="LDAP_Configuration"
+			dn="#['cn=Test User,ou=DevOpsGroup,' ++ attributes.queryParams.dn]"
+			doc:name="Delete User Entry from LDAP Directory" />
+
+		<ldap:delete config-ref="LDAP_Configuration"
+			dn="#['ou=DevOpsGroup,' ++ attributes.queryParams.dn]"
+			doc:name="Delete Group Entry from LDAP Directory" />
+
+		<ee:transform doc:name="DataWeave to set Payload indicating flow completed">
+			<ee:message>
+				<ee:set-payload><![CDATA[%dw 2.0
+output application/json
+---
+{
+	result : "Flow Successfully Completed"
+}
+]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+	</flow>
+
+
+</mule>
+----
+
+[[example-use-case2]]
+=== Example Use Case 2 with LDAP Connector
+
+A custom trust store can be setup to tell which servers are allowed to communicate to.
+
+Extended configuration parameters can be used for this to specify a custom trust store.
+
+The same Use Case 1 (above) can be used to execute this except for the configuration part of LDAP
+connector which should now use TLS configuration.
+
+Find below the XML configuration snippet of LDAP Connector which uses TLS configuration and update
+the Use Case 1 XML file (above) with it.
+
+[source,xml,linenums]
+----
+<ldap:config name="LDAP_Configuration" doc:name="LDAP Configuration">
```

Comment:
Something worth mentioning:

This configuration still works, and will override the new Security configuration to avoid breaking backward compatibility, but using the new feature is preferred.